contracts/message-bus/MessageInbox.sol

```diff
@@ -75,20 +75,24 @@ contract MessageInbox is MessageBox, Proof {
     /**
      * @notice Setup message inbox.
      *
-     * @dev Function requires:
-     *          - inboundChannelIdentifier must be zero
-     *          - metachainId must not be zero
-     *          - messageOutbox address must not be zero
-     *          - verifyingAddress must not be zero
-     *          - stateRootProvider must not be zero
-     *
      * @param _metachainId Metachain identifier.
      * @param _messageOutbox MessageOutbox contract address.
      * @param _outboxStorageIndex Storage index of outbox mapping in
      *                            MessageOutbox contract.
      * @param _stateRootProvider State root provider contract address.
      * @param _maxStorageRootItems Defines how many storage roots should be
      *                             stored in circular buffer.
+     *
+     * \pre Setup can only be called once.
+     * \pre `_metachainId` must not be zero.
+     * \pre `_messageOutbox` address must not be zero.
+     * \pre `_stateRootProvider` must not be zero.
+     * \pre Satisfies all the pre conditions of setup of proof contract.
```

Comment:
Should we abandon this pattern and try to mention all pre and post conditions instead here also? In this case no pre-conditions I assume is going to be added on top of what is already there.
/cc @deepesh-kn

Reply:
Let try with what we discussed today, where we specify the function calls.

Reply:
Fixed.